### PR TITLE
CI: Update the Fedora Container to F34

### PR DESCRIPTION
**Pull request by marcusmueller**
this has the main advantage of bringing UHD4, so we can test more of gr-uhd.

It's uncertain whether we'd want to backport this right now – F33 is still supported and maybe it's more important to test maint-3.9 with the current-at-release system than with UHD4.
It might thus make sense to hold off until the next 3.9.X release.

### CODE REVIEW

**Comment by willcode** [comment]
Agree we should keep F33 for now. I build F34 myself and will notice if something terrible happens (but with UHD4). F35 will be out in Oct.

**Comment by mbr0wn** [comment]
Why can't we have both?

**Comment by marcusmueller** [comment]
I thought I had answered, but doesn't look that way:

> Why can't we have both?

@mbr0wn  I thought we only hate finite CPU resources. I'd rather have builds that finish sooner than multiple highly correlated builds; but that was just a wild guess on my side!

**Comment by mbr0wn** [comment]
> @mbr0wn I thought we only hate finite CPU resources. I'd rather have builds that finish sooner than multiple highly correlated builds; but that was just a wild guess on my side!

Builds run in parallel; I don't think this will delay anything during builds.

**Comment by mbr0wn** [comment]
> Are we switching back from the `/ci:` to the `/ci-` images? I thought that was only for Debian 10 which has a mysterious failure and didn't get updated.

I thought so too.
> 
> More Fedora people should be on 34 now, so good to move 3.9 testing up.

:+1:

**Comment by marcusmueller** [comment]
the colon seems to get automagically converted to `-` if there's no ambiguity arising from an attached tag name, so it's not something you can rely on. That's why I removed it, after it took me a while to figure out.

**Comment by willcode** [comment]
These are two completely different sets of images, build from different Dockerfiles, aren't they?

**Comment by marcusmueller** [comment]
> These are two completely different sets of images, build from different Dockerfiles, aren't they?

I don't think so, but might be confused. That's what the gnuradio dockerhub repo entails:

Untitled

**Comment by marcusmueller** [comment]
You might be right, the ci:something might be tags of the ci container. https://hub.docker.com/repository/docker/gnuradio/ci

**Comment by willcode** [comment]
Yes, they are tags.

**Comment by marcusmueller** [comment]
which means I need to push the things I pushed to ci-something as ci:something,  fix the yml here (and PR #4981).

File: .github/workflows/make-test.yml [hunk 1]
```diff
@@ -54,8 +54,8 @@ jobs:
           - distro: 'Ubuntu 20.04'
             containerid: 'gnuradio/ci:ubuntu-20.04-3.9'
             cxxflags: -Werror -Wno-error=invalid-pch
-          - distro: 'Fedora 33'
-            containerid: 'gnuradio/ci:fedora-33-3.9'
+          - distro: 'Fedora 34'
+            containerid: 'gnuradio/ci-fedora-34-3.9:1.0'
```
Comment (mbr0wn) [inline .github/workflows/make-test.yml:58]:
```suggestion
          - distro: 'Fedora 33'
            containerid: 'gnuradio/ci:fedora-33-3.9'
          - distro: 'Fedora 34'
            containerid: 'gnuradio/ci:fedora-34-3.9'            
```

File: .github/workflows/pkg-fedora.yml [hunk 2]
```diff
@@ -26,8 +26,8 @@ jobs:
         # descriptive name, and one key 'containerid' with the name of the
         # container (i.e., what you want to docker-pull)
         include:
-          - distro: 'Fedora 33'
-            containerid: 'gnuradio/ci:fedora-33-3.9'
+          - distro: 'Fedora 34'
+            containerid: 'gnuradio/ci-fedora-34-3.9:1.0'
```
Comment (mbr0wn) [inline .github/workflows/pkg-fedora.yml:30]:
```suggestion
          - distro: 'Fedora 33'
            containerid: 'gnuradio/ci:fedora-33-3.9'
          - distro: 'Fedora 34'
            containerid: 'gnuradio/ci:fedora-34-3.9' 
```

**Comment by mbr0wn** [comment]
We had different ways of doing Docker builds. Now we use autobuilds, and they all have the same tag. I added some suggestions in the review.

**Comment by marcusmueller** [comment]
@mbr0wn  how do these autobuilds work? What do I need to do to trigger one?

**Comment by willcode** [comment]
Dockerhub will find the change in Github and redo the build automatically.

**Comment by mbr0wn** [comment]
I thought containers get autobuilt when you push to master? @noc0lour set it up.

**Comment by marcusmueller** [comment]
> I thought containers get autobuilt when you push to master? @noc0lour set it up.

Not anymore:
https://github.com/gnuradio/gnuradio/issues/5051

Opting to not to shave the fix-autobuilds yak right now. So, also, makes no difference whether we use ci:something or ci-anything as all is transient.

File: .github/workflows/pkg-fedora.yml (hunk 2 above)
Comment (marcusmueller) [reply .github/workflows/pkg-fedora.yml:30]: These tags do not exist, so sadly not possible :(

**Comment by willcode** [comment]
You mean back to build-container-on-the-fly?

**Comment by noc0lour** [comment]
> > I thought containers get autobuilt when you push to master? @noc0lour set it up.
> 
> Not anymore:
> #5051
> 
> Opting to not to shave the fix-autobuilds yak right now. So, also, makes no difference whether we use ci:something or ci-anything as all is transient.

It's easier to trace if you have one repo with multiple tags vs a lot of new image repositories. Just my two cents.

File: .github/workflows/pkg-fedora.yml (hunk 2 above)
Comment (mbr0wn) [reply .github/workflows/pkg-fedora.yml]: They do now.

File: .github/workflows/make-test.yml (hunk 1 above)
Comment (mbr0wn) [reply .github/workflows/make-test.yml]: Booyah.

**Review by mbr0wn — APPROVED** [review]

**Comment by mbr0wn** [comment]
Meh weird github auto-close. Will replace this.

**Pull request closed without merging**